pwndbg/dbg/gdb/__init__.py

```diff
@@ -573,21 +595,40 @@ def create_value(
 
     @override
     def symbol_name_at_address(self, address: int) -> str | None:
-        import pwndbg.gdblib.symbol
+        from pwndbg.dbg.gdb.symbol import resolve_addr
 
-        return pwndbg.gdblib.symbol.get(address) or None
+        return resolve_addr(address) or None
 
     @override
-    def symbol_address_from_name(self, name: str, prefer_static: bool = False) -> int | None:
-        import pwndbg.gdblib.symbol
-
+    def lookup_symbol(
+        self,
+        name: str,
+        *,
+        prefer_static: bool = False,
+        type: pwndbg.dbg_mod.SymbolLookupType = pwndbg.dbg_mod.SymbolLookupType.ANY,
+        objfile_endswith: str | None = None,
+    ) -> pwndbg.dbg_mod.Value | None:
+        from pwndbg.dbg.gdb.symbol import Domain
+        from pwndbg.dbg.gdb.symbol import lookup_symbol
+
+        domain = {
+            pwndbg.dbg_mod.SymbolLookupType.ANY: Domain.ANY,
+            pwndbg.dbg_mod.SymbolLookupType.VARIABLE: Domain.VARIABLE,
+            pwndbg.dbg_mod.SymbolLookupType.FUNCTION: Domain.FUNCTION,
+        }[type]
```

Comment:
Why do we need this mapping?

If it is really needed, can we do if/elif/elif/else